Fix quantal stdp test for numpy

mschmidt87 opened the pull request:
This pull request addresses #680  and contains the following changes:

- Convert the float variables in the test to integers when they are used as indices to numpy arrays. I decided to do this rather then defining them as integers in the first place because a) defining times as floats seems plausible to me and b) the conversion of `t_tot` had been implemented in this test at a different location earlier.
- Remove two parameter dictionaries that were not used in the script.
- Make travis download the latest release of numpy via pip.

heplesser commented on pynest/nest/tests/test_quantal_stp_synapse.py:
```diff
@@ -101,15 +97,16 @@ def test_QuantalSTPSynapse(self):
-        vm.shape = (n_trials, t_tot)
-        vm_reference.shape = (n_trials, t_tot)
+        vm.shape = (n_trials, int(t_tot))
+        vm_reference.shape = (n_trials, int(t_tot))
```
Using `int(t_tot)` here to dimension the result array is not a good idea, since you implicitly require that the voltmeter records at 1 ms intervals. A cleaner implementation would be (not 100% sure of spelling of assertions)
```python
self.assert_equal(len(vm) % n_trials, 0)
n_steps = int(len(vm) / n_trials)    # int to be absolutely Py3-safe
vm.shape = (n_trials, n_steps)
...
```

heplesser commented on pynest/nest/tests/test_quantal_stp_synapse.py:
```diff
@@ -101,15 +97,16 @@ def test_QuantalSTPSynapse(self):
+        vm_reference.shape = (n_trials, int(t_tot))
 
         vm_mean = numpy.array([numpy.mean(vm[:, i])
                                for i in range(int(t_tot))])
```
I think you can avoid the list comprehension here, NumPy supports row- and column-wise means.

mschmidt87 commented:
Thank you for your feedback. I removed the commit touching .travis.yml from the PR and addressed your comments. I furthermore removed `t_plot` from the script because it was only used for computing the error and I didn't see a reason to not take the entire dataset into account here.
